docs/recommendations.md

```diff
@@ -16,3 +16,51 @@ Always bind to localhost rather than to 0.0.0.0 or any interface, unless there i
 Binding to 0.0.0.0 or "all" interfaces can make your application respond on all current and future network interfaces.
 
 Issue: ([#19](https://github.com/open-telemetry/sig-security/issues/19#issue-1926445623))
+
+## Container images
+
+### Use the smallest image possible
+While the size of an image doesn't matter for the runtime, we are responsible for everything that is in the image. If you are delivering a statically linked binary, you can probably build a `scratch` image for production purposes. It might be a good idea to provide a second set of images for debugging purposes, such as ones including a shell and networking utilities.
+
+If you need to include files in the image, such as root certificates, use the builder pattern to obtain the files and copy only those into the final image. Example:
+
+```Dockerfile
+FROM alpine:3.20 as certificates
+RUN apk --no-cache add ca-certificates
+
+FROM scratch
+COPY --from=certificates /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+COPY mybinary mybinary
+```
+
+### Use a non-privileged user
+Under some circumstances, your image might be executed as a privileged user by default. In others, like a hardened Kubernetes distribution, your container is forced to run using a non-privileged account and bugs might appear as the image wasn't built with that in mind. It's recommended to set a high user ID when building the image, to ensure the process always runs as a non-privileged user. This will not only make it more secure by default but will help you uncover issues before they affect security-conscious users.
+
+Here's how to do it:
+
+```Dockerfile
+USER 65532:65532
```

Comment:
There could be some more things here, e.g. you likely need to create the user in the local environment (see https://stackoverflow.com/questions/67261873/building-docker-image-as-non-root-user)

Additionally, when using shared-volumes (e.g. docker-image as a development tool), there's a set of things you can do so that files generated use the local user, e.g. https://github.com/open-telemetry/weaver/blob/main/docs/docker-guide.md#basic-usage---otel-semconv-codegen


Finally - Many docker images that allow custom users, actually have "flags", even ENV flags, which will allow you to control the UID/GID and username in the container, e.g. if you want to line it up with the local users and shared volumes. (can't find link just now).

Reply:
> you likely need to create the user in the local environment

I'm not quite sure you need to create the user: the UID/GID is being set to an explicit number. The `useradd` command would only add the user to the container, which would only serve to map the UID/GID to a name. This wouldn't propagate back to the host. Perhaps I'm missing the point about this?

 > there's a set of things you can do so that files generated use the local user

That's mostly for an end-user, right? Those would be good things for SIGs to document for their users. I'll mention that. Is there something to be added to the Dockerfile itself?

Reply:
Created https://github.com/open-telemetry/sig-security/issues/73 to track.